source/slang/slang-lower-to-ir.cpp

```diff
@@ -8549,6 +8551,18 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
             }
         }
 
+        // Add modifier to constructors so they are aware of the parent IRType.
+        // This follows the lowering logic of ensureAllDeclsRec->{visitAggTypeDecl, visitFunctionDeclBase}
+        if(isStruct)
+        {
+            for (auto ctor : decl->getMembersOfType<ConstructorDecl>())
+            {
+                auto parentStructTypeModifier = this->context->astBuilder->create<ParentStructTypeModifier>();
```

Comment:
You can simply maintain a side band dictionary in the ir lowering context to keep track of struct types and their ctors.